djangosaml2idp/processors.py

```diff
@@ -47,14 +49,20 @@ def _get_nameid_opaque(cls, user_id: str, salt: bytes = b'', *args, **kwargs) ->
         return opaque.hexdigest()
 
     @classmethod
-    def get_nameid_persistent(cls, user_id: str, user: User, sp_entityid: str = '', idp_entityid: str = '') -> str:
+    def get_nameid_persistent(cls, user_id: str, sp: ServiceProvider, user: settings.AUTH_USER_MODEL) -> str:
```

Comment:
Is it necessary to pass the user instance as well? It is only used for lookup, which the user_id can be used for as well.

Reply:
Actually, yes: depending on the SP, the field used for user_id could be anything, so it's a lot simpler to just pass the user instance.